service/src/main/java/bio/terra/pipelines/service/PipelinesService.java

```diff
@@ -259,4 +263,31 @@ protected ParameterTypeDefinition mapInputTypeToCbasParameterType(PipelineInputT
           .type(ParameterTypeDefinition.TypeEnum.ARRAY);
     };
   }
+
+  /**
+   * Prepare a list of CBAS WorkflowOutputDefinitions for a given pipeline and WDL method name.
+   *
+   * @param pipelineOutputDefinitions
+   * @param wdlMethodName
+   * @return
+   */
+  public List<WorkflowOutputDefinition> prepareCbasWorkflowOutputRecordUpdateDefinitions(
+      List<PipelineOutputDefinition> pipelineOutputDefinitions, String wdlMethodName) {
+    return pipelineOutputDefinitions.stream()
+        .map(
+            pipelineOutputDefinition -> {
+              String outputName = pipelineOutputDefinition.getWdlVariableName();
+              return new WorkflowOutputDefinition()
+                  .outputName("%s.%s".formatted(wdlMethodName, outputName))
+                  .outputType(
```

Comment:
shouldnt this be based on what the type the pipelineOutputDefinition is?

Reply:
yes, but currently there's only one type (FILE) so until we added others I was keeping it simple. when we switch to using the type enum for outputs, then we'll want to add that conditional logic